<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1380</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3DViewer 1.0</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="OGL" name="OGLWindow">
    <property name="geometry">
     <rect>
      <x>387</x>
      <y>2</y>
      <width>1000</width>
      <height>1000</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>7</y>
      <width>361</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Translate Object</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QScrollBar" name="scroll_translate_y">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>70</y>
      <width>272</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>69</y>
      <width>37</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y Axis</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>99</y>
      <width>37</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z Axis</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>40</y>
      <width>38</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X Axis</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="scroll_translate_x">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>41</y>
      <width>272</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QScrollBar" name="scroll_translate_z">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>100</y>
      <width>272</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>14</x>
      <y>160</y>
      <width>361</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Object</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QScrollBar" name="scroll_rotate_z">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>254</y>
      <width>272</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-180</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>194</y>
      <width>38</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X Axis</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>253</y>
      <width>37</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z Axis</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="scroll_rotate_x">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>195</y>
      <width>272</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-180</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>223</y>
      <width>37</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y Axis</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="scroll_rotate_y">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>224</y>
      <width>272</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-180</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>312</y>
      <width>361</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale Object</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QScrollBar" name="scroll_scale">
    <property name="geometry">
     <rect>
      <x>49</x>
      <y>339</y>
      <width>272</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>140</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>292</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>366</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>338</y>
      <width>37</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Ratio</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>396</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Projection Type</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_projection_par">
    <property name="geometry">
     <rect>
      <x>292</x>
      <y>394</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Parallel</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_proj_type</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_projection_cent">
    <property name="geometry">
     <rect>
      <x>214</x>
      <y>394</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Central</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_proj_type</string>
    </attribute>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>426</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_translate_x">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>38</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_translate_y">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>70</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_translate_z">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>100</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_rotate_z">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>254</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-180</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_rotate_y">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>224</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-180</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_rotate_x">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>192</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-180</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spin_scale">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>337</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.100000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="stepType">
     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>456</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Edges</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_edges_color">
    <property name="geometry">
     <rect>
      <x>64</x>
      <y>450</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose Color</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>186</x>
      <y>456</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Type</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_edges_solid">
    <property name="geometry">
     <rect>
      <x>227</x>
      <y>454</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Solid</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_edges_type</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_edges_dashed">
    <property name="geometry">
     <rect>
      <x>292</x>
      <y>454</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Dashed</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_edges_type</string>
    </attribute>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>501</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Thickness</string>
    </property>
   </widget>
   <widget class="QSlider" name="slider_edges_thickness">
    <property name="geometry">
     <rect>
      <x>87</x>
      <y>497</y>
      <width>271</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>528</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>556</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertices</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_vertices_color">
    <property name="geometry">
     <rect>
      <x>64</x>
      <y>549</y>
      <width>316</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose Color</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>596</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Display Method</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_verctices_display_none">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>592</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>None</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_vertices_display</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_verctices_display_circle">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>592</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Circle</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_vertices_display</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_verctices_display_square">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>592</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Square</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_vertices_display</string>
    </attribute>
   </widget>
   <widget class="QSlider" name="slider_vertices_size">
    <property name="geometry">
     <rect>
      <x>87</x>
      <y>632</y>
      <width>271</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>636</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Size</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_gif_record">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>885</y>
      <width>181</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Capture GIF</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_image_save">
    <property name="geometry">
     <rect>
      <x>199</x>
      <y>885</y>
      <width>181</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Image</string>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>670</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_7">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>860</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_8">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>919</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_open_file">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>950</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Open File</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_file_name">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>954</y>
      <width>260</width>
      <height>21</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>700</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Background</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_background_color">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>694</y>
      <width>290</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose color</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_init_view_settings">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>756</y>
      <width>373</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Init Viewer Settings</string>
    </property>
   </widget>
   <widget class="Line" name="line_9">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>731</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_10">
    <property name="geometry">
     <rect>
      <x>7</x>
      <y>794</y>
      <width>371</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_vert_num">
    <property name="geometry">
     <rect>
      <x>128</x>
      <y>825</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_edges_num">
    <property name="geometry">
     <rect>
      <x>309</x>
      <y>825</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_19">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>826</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of vertices:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_20">
    <property name="geometry">
     <rect>
      <x>202</x>
      <y>826</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of edges:</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OGL</class>
   <extends>QOpenGLWidget</extends>
   <header>ogl.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>spin_translate_x</tabstop>
  <tabstop>spin_translate_y</tabstop>
  <tabstop>spin_translate_z</tabstop>
  <tabstop>spin_rotate_x</tabstop>
  <tabstop>spin_rotate_y</tabstop>
  <tabstop>spin_rotate_z</tabstop>
  <tabstop>spin_scale</tabstop>
  <tabstop>radioButton_projection_cent</tabstop>
  <tabstop>radioButton_projection_par</tabstop>
  <tabstop>pushButton_edges_color</tabstop>
  <tabstop>radioButton_edges_solid</tabstop>
  <tabstop>radioButton_edges_dashed</tabstop>
  <tabstop>slider_edges_thickness</tabstop>
  <tabstop>pushButton_vertices_color</tabstop>
  <tabstop>radioButton_verctices_display_none</tabstop>
  <tabstop>radioButton_verctices_display_circle</tabstop>
  <tabstop>radioButton_verctices_display_square</tabstop>
  <tabstop>slider_vertices_size</tabstop>
  <tabstop>pushButton_gif_record</tabstop>
  <tabstop>pushButton_image_save</tabstop>
  <tabstop>pushButton_open_file</tabstop>
  <tabstop>lineEdit_file_name</tabstop>
  <tabstop>pushButton_background_color</tabstop>
  <tabstop>pushButton_init_view_settings</tabstop>
  <tabstop>lineEdit_vert_num</tabstop>
  <tabstop>lineEdit_edges_num</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup_edges_type"/>
  <buttongroup name="buttonGroup_vertices_display"/>
  <buttongroup name="buttonGroup_proj_type"/>
 </buttongroups>
</ui>
